src/main/java/nl/hsac/fitnesse/util/AllArgScenarioUtils.java

```diff
@@ -0,0 +1,32 @@
+package nl.hsac.fitnesse.util;
+
+import fitnesse.slim.StopTestSlimException;
+import fitnesse.testsystems.TestExecutionException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AllArgScenarioUtils {
+    /**
+     * Method allows to parse argument list provided with @{*} symbol in all arg table
+     * @param args list of variables and their values
+     * @return parsed map of input arguments which can be used to easily map values in fixture
+     * @throws StopTestSlimException when number of parameters don't match expected value
+     */
+    public static HashMap<String,String> parseFitNesseArgsList(String[] args) throws StopTestSlimException {
+        HashMap<String,String> argsParsed = new HashMap<>();
+        ArrayList<String> tmpList = (ArrayList<String>) Arrays.stream(args).collect(Collectors.toList());
+        if(Math.floorMod(tmpList.size(),2) != 0) {
+            throw new StopTestSlimException("Incorrect number of arguments in class constructor. Please check your test syntax. The only attribute that is allowed in this constructor is @{*}");
```

Comment:
This message does not seem appropriate in a util class, given the check you perform. I believe you just want an even number of elements in the array.

I might be wrong, but this whole method seems unnecessarily complex and inefficient. Why don't you just check the `args` directly and loop over it to fill the map? Furthermore I recommend filling a `LinkedHashMap` (so the element order is preserved in case someone cares).

Reply:
To be honest I spend like 2 minutes writing this method and did not think much about implementation as it was meant to be only an example. I refactored it in TestFixture.class